### PR TITLE
[IMP] mail, *: simplify discuss template (no templateGetter)

### PR DESCRIPTION
**Pull request by alexkuhn**
This commit simplifies discuss template by putting record accessors in the context of template.

*: calendar, hr_holidays, im_livechat, sms, snailmail,
   website_livechat

Task-3055022

Enterprise: https://github.com/odoo/enterprise/pull/33575

### CODE REVIEW

File: addons/mail/static/src/components/chat_window_header/chat_window_header.xml [hunk 1]
```diff
@@ -2,76 +2,75 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.ChatWindowHeaderView" owl="1">
-        <t t-if="chatWindowHeaderView">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/chat_window_header/chat_window_header.xml:5]: remove `</t>`

File: addons/mail/static/src/model/model_manager.js [hunk 2]
```diff
@@ -402,7 +402,7 @@ export class ModelManager {
             }
             Object.defineProperty(
                 ModelComponent.prototype,
-                definition.get('templateGetter') ? definition.get('templateGetter') : 'record',
+                definition.get('templateGetter') || 'record',
```
Comment (seb-odoo) [inline addons/mail/static/src/model/model_manager.js:405]: remove support

File: addons/mail/static/src/model/model_manager.js [hunk 3]
```diff
@@ -1319,7 +1319,39 @@ export class ModelManager {
...
+                }
+                for (const name of definition.get('recordGetters').keys()) {
+                    Object.defineProperty(model.__messagingComponent.prototype, name, {
+                        get() {
+                            return this.props.record[name];
+                        },
+                    });
+                }
             }
+
```
Comment (seb-odoo) [inline addons/mail/static/src/model/model_manager.js:1354]: no need for empty line

File: addons/mail/static/src/components/composer/composer.xml [hunk 4]
```diff
@@ -2,167 +2,165 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.ComposerView" owl="1">
-        <t t-if="composerView">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer/composer.xml:5]: remove `</t>`

File: addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.xml [hunk 5]
```diff
@@ -2,33 +2,32 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.DiscussSidebarView" owl="1">
-        <t t-if="discussSidebarView">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.xml:5]: remove `</t>`

File: addons/mail/static/src/components/discuss_sidebar_category/discuss_sidebar_category.xml [hunk 6]
```diff
@@ -2,50 +2,49 @@
 
 <templates xml:space="preserve">
     <t t-name="mail.DiscussSidebarCategory" owl="1">
-        <t t-if="category">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/discuss_sidebar_category/discuss_sidebar_category.xml:5]: remove `</t>`

File: addons/mail/static/src/components/thread_view/thread_view.xml [hunk 7]
```diff
@@ -2,60 +2,59 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.ThreadView" owl="1">
-        <t t-if="threadView">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/thread_view/thread_view.xml:5]: remove `</t>`

File: addons/mail/static/src/components/composer/composer.xml (hunk 4 above)
Comment (alexkuhn) [reply addons/mail/static/src/components/composer/composer.xml:5]: Looks properly removed, please note there's a local `t-name` in the file, so this `t-name="mail.ComposerView"` ends at 2/3rd of the file

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/105099

**Pull request closed without merging**